runelite-mixins/src/main/java/net/runelite/mixins/RSClientMixin.java

```diff
@@ -106,6 +113,12 @@
 	@Inject
 	private static boolean interpolateObjectAnimations;
 
+	@Inject
+	private static RSPlayer[] oldPlayers = new RSPlayer[32768];
```

Comment:
Do we really need to statically allocate an array of size 32768 * 64?  This is 262kb of a mostly sparse array.

Guava has a Maps.difference function that can be used if you want to use an int indexed map to find differences.
Though I suppose using such would also make more garbage.

Reply:
I am mostly indifferent between this and a hash map. This seems simplier.

Reply:
I believe the player array actually only needs to have 2048 entries, so we can reduce it in size a bit